cpp/oneapi/dal/graph/service_functions_test.cpp

```diff
@@ -0,0 +1,166 @@
+#include "oneapi/dal/test/engine/common.hpp"
+
+#include "oneapi/dal/graph/service_functions.hpp"
+#include "oneapi/dal/graph/undirected_adjacency_vector_graph.hpp"
+#include "oneapi/dal/io/graph_csv_data_source.hpp"
+#include "oneapi/dal/io/load_graph.hpp"
+
+namespace oneapi::dal::graph::test {
+
+namespace dal = oneapi::dal;
+
+using my_graph_type = dal::preview::undirected_adjacency_vector_graph<>;
+const dal::preview::load_graph::descriptor<dal::preview::edge_list<int32_t>, my_graph_type>
+    desc;
+
+auto get_data_source (const std::string& name) {
+	const std::string name_path = "cpp/oneapi/dal/graph/test_data/" + name;
+	const dal::preview::graph_csv_data_source ds(name_path);
+	return ds;
+}
+
+auto check_neighbors (const std::string& name, 
+	const dal::preview::vertex_size_type<my_graph_type> *correct_neigh,
+	const dal::preview::vertex_size_type<my_graph_type> vertices_num) {
+	int correct_neigh_num = 0;
+	dal::preview::vertex_size_type<my_graph_type> neigh_index = 0;
+	const auto my_graph = dal::preview::load_graph::load(desc, get_data_source(name));
+	for (dal::preview::vertex_size_type<my_graph_type> j = 0; j < vertices_num; j++) {
+		const auto neigh = dal::preview::get_vertex_neighbors(my_graph, j);
+		for (auto i = neigh.first; i != neigh.second; ++i) {
+            if (*i == correct_neigh[neigh_index]) correct_neigh_num++;	
+        	neigh_index++;
+        }
+    }
+    return correct_neigh_num;
+}
+
+TEST_CASE("throws if input file is empty") {
+	REQUIRE_THROWS_WITH(dal::preview::load_graph::load(desc, get_data_source("graph0.csv")),"Empty edge list");
```

Comment:
I suggest to not use `load_graph` in such simple test cases. I consider this test as **unit**, so the only one isolated component should be tests.

Right approach: Do not use `load_graph`, create graph from the data that is already in memory.

Reply:
What about compile-time file loading with `#include` directive? Is it a valid option?

Reply:
@KulikovNikita, the main problem with that test is that it does not test `graph/service_functions`, but tests `load_graph`. This test checks if empty CSV is correctly handled by data source as an empty graph. That kind of tests must be placed in `io/load_graph` submodule.

Possible approaches for data source testing:
1. Lightweight (smoke) tests on very small datasets
   - Have small datasets embedded right into the source code using C-style static array, `std::array`, etc. I don't very like `#include` idea, because the small datasets can be embedded without lose of readability.
   - Use [fixtures](https://github.com/catchorg/Catch2/blob/devel/docs/test-fixtures.md) to write the data to some file on
      fixture initialization.
   - Read that file using `load_graph` and test properties of the object loaded
2. Heavy tests
    - Don't run in the public CI
    - Have special large files stored in special location, e.g., `DATASETS_ROOT`, don't put them to repository

Reply:
Thank you for your comments. I'll edit the tests according to them.
I have to use `load_graph` in the service functions tests because `load_graph::load` is currently the only way to create a graph.